backend/app/core/storage.py

```
@@ -0,0 +1,99 @@
+# Configure basic logging for the mock storage
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class StorageInterface(abc.ABC):
+    """
+    Abstract Base Class for storage operations.
+    """
+
+    @abc.abstractmethod
+    async def upload_file(
+        self, file_content: bytes, destination_blob_name: str, content_type: Optional[str] = None
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)